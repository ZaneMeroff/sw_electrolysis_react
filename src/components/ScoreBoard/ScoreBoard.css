.scoreboard-outer-container {
  align-items: center;
  display: flex;
  width: 345px;
}

.game-button {
  background-color: #313131;
  border: 2px dotted #FFF;
  color: #FFF;
  font-family: 'Press Start 2P', cursive;
  font-size: 70%;
  height: auto;
  margin: 5px 0px;
  outline: none;
  padding: 4px;
  width: 150px;
}

.game-button:hover,
.game-button:focus {
  background-color: #FFF;
  border: 2px dotted #313131;
  color: #313131;
  cursor: pointer;
}

.round-display {
  color: #FFF;
  font-family: 'Press Start 2P', cursive;
  font-size: 70%;
  margin: 8px 30px 8px 20px;
}

.round-moves-container {
  align-items: flex-start;
  display: flex;
  flex-direction: column;
}

.score-board-button-container {
  display: flex;
  flex-direction: column;
  margin-left: 10px;
  justify-content: space-between;
}

.hidden {
  display: none;
}

.disabled {
  pointer-events: none;
}
