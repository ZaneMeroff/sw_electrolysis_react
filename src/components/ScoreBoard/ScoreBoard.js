import React from 'react';
import './ScoreBoard.css';

export const ScoreBoard = props => {

  return (
    <div className="scoreboard-outer-container">
      <div className='round-moves-container'>
        <h3 className='round-display'>ROUND: {props.roundCount}</h3>
        <h3 className='round-display'>MOVES: {props.playerClickDisplay}/{props.roundCount}</h3>
      </div>
      <div className={props.disableScoreBoard ? 'disabled score-board-button-container' : 'score-board-button-container'}>
        <button aria-label='start game' onClick={props.startGame} className='game-button'>{props.randomNumArray.length !== 0 ? 'NEW GAME' : 'START GAME'}</button>
        <button aria-label='toggle view rules' onClick={props.onViewRulesClick} className='game-button'>{props.rulesVisible ? 'HIDE RULES' : 'VIEW RULES'}</button>
      </div>
    </div>
  );

}
