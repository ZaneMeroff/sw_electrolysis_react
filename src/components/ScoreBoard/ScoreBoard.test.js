import React from 'react';
import { ScoreBoard } from './ScoreBoard';
import { shallow } from 'enzyme';

describe('ScoreBoard', () => {

  let wrapper;

  beforeEach(() => {
    const mockProps = {
     roundCount: 0,
     startGame: jest.fn(),
     playerClickDisplay: 0,
     randomNumArray: [1, 0, 3, 1, 2, 0, 2, 1, 0, 2]
    }
    wrapper = shallow(<ScoreBoard {...mockProps} />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
