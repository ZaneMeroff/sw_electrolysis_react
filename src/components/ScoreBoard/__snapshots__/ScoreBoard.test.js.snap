// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreBoard should match the snapshot 1`] = `
<div
  className="scoreboard-outer-container"
>
  <div
    className="round-moves-container"
  >
    <h3
      className="round-display"
    >
      ROUND: 
      0
    </h3>
    <h3
      className="round-display"
    >
      MOVES: 
      0
      /
      0
    </h3>
  </div>
  <div
    className="score-board-button-container"
  >
    <button
      aria-label="start game"
      className="game-button"
      onClick={[MockFunction]}
    >
      NEW GAME
    </button>
    <button
      aria-label="toggle view rules"
      className="game-button"
    >
      VIEW RULES
    </button>
  </div>
</div>
`;
